source/common/http/http2/codec_impl.cc

```diff
@@ -25,6 +25,26 @@ namespace Envoy {
 namespace Http {
 namespace Http2 {
 
+class Http2ResponseCodeDetailValues {
+  const absl::string_view NgHttp2ErrHttpHeader = "http2.nghttp2_err_http_header";
```

Comment:
mind looking at the nghttp2 explanations and seeing if we can put something more readable here and/or in comments?
I suspect one is for a particular header being invalid, one is for violating the spec either in the header block or at large.